Thermal Comfort Model Documentation Corrections

RKStrand opened the pull request:
Pull request overview
---------------------
This pull request corrects many problems with the thermal comfort model documentation sections for the Fanger, Pierce, and KSU models as requested by the defect description.  Equations were rechecked and corrected when discrepancies were found.  Now, everything lines up with what is in the code.  Lots of changes for the better.  This was requested by GitHub Issue #2996.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue #2996 

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

RKStrand committed:
Correction of many problems

This commit corrects many problems with the thermal comfort model
documentation sections for the Fanger, Pierce, and KSU models as
requested by the defect description.  Equations were rechecked and
corrected when discrepancies were found.  Now, everything lines up with
what is in the code.  Lots of changes for the better.

RKStrand commented:
@Myoldmopar @michael-okeefe @mjwitte Would one of you be willing to review this?  Thanks, Rick

michael-okeefe commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -336,163 +364,224 @@ \subsubsection{Pierce Two-Node Model Nomenclature List}\label{pierce-two-node-mo
 HeatFlow = (CoreTemp-SkinTemp)*(5.28 + 1.163*SkinBloodFlow)
 
-Where
+where 5.28 is the average body tissue conductance in W/m\(^{2}\)•°C and 1.163 is the thermal capacity of blood in W•h/L•°C
```
Looks like some of the unicode characters may have gotten scrambled up. Maybe:

`W/m\(^2 \mdot \) °C`

michael-okeefe replied:
Sorry that should be:

`W/m\(^2 \cdot \) °C`

I've been working with mass flow rates too much. `\cdot` is center-dot. You could also use `\bullet` if you wanted to match the original Word closer.

RKStrand replied:
@michael-okeefe Thanks for the review--I'll fix that as soon as possible.

RKStrand replied:
@michael-okeefe By the way, I have been building the PDF locally on my computer to make sure that things are formatted correctly.  So, hopefully once it's up in GitHub things should be good to go other than any typos, etc.

michael-okeefe replied:
Awesome. Sounds great!

Myoldmopar replied:
@RKStrand This still needs to be addressed but once that is fixed, this can be merged in.

RKStrand replied:
@Myoldmopar Yes, I will correct this very soon along with equation you mentioned and get this checked back in.  Will keep you posted...

RKStrand replied:
@michael-okeefe @Myoldmopar Ok, this issue is fixed now.  Changed the degrees C to K...because despite many attempts I couldn't be both of the degree symbols to show up.  Weird but using just K works and is the same thing.

michael-okeefe replied:
Yeah, LaTeX is definitely a black art. In the past, I've used:

`$^{o}$C` or `$^{\circ}$C` to get the "degree" symbol to print although there are more eloquent ways. `K` is fine with me although I'll defer to Edwin.

Great work!

michael-okeefe reviewed (changes requested):
Looks pretty good. I didn't build it to look at the output PDF -- I would recommend doing that if you haven't already to make sure everything is looking OK. Otherwise, just one minor glitch found with some unicode characters that should be a quick fix.

Myoldmopar commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -1,22 +1,24 @@
+There are two personal variables that influence the condition of thermal comfort: the thermal resistance of the clothing (I\(_{cl}\)), and the metabolic rate (H/A\(_{Du}\)).
 
-The metabolic rate (H/A\(_{Du}\)), is a measure of the internal heat production rate of an occupant (H) w/hr. in per unit of ``Dubois'' body surface area (A\(_{Du}\)) in units of m\(^{2}\). The DuBois body surface area is given by :
+The thermal resistance of the clothing (I\(_{cl}\)) is measured in units of ``clo.'' The 1985 ASHRAE Handbook of Fundamentals (ASHRAE 1985) suggests multiplying the summation of the individual clothing items clo value by a factor of 0.82 to determine the overall clo value for clothing ensembles.
```
In general, I'd suggest in the future using dollar signs to drop into and out of inline math mode.  Using the escaped parentheses is more bulky, and difficult to read, especially when there are other parentheses around.  No need to change all these here, I know they are currently all over the docs, just a suggestion for the future.

RKStrand replied:
@Myoldmopar Thanks for the suggestion--I will definitely keep that in mind in any future documentation work.  I admit that I don't know all the ins and outs of what is best--I was just trying to figure out how to get everything to look correct and I borrowed stuff from what was already there.  Apparently I used the "bad" examples!

Myoldmopar commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -50,21 +52,21 @@ \subsection{Background on Thermal Comfort Models}\label{background-on-thermal-co
 \caption[]{General Nomenclature list for Thermal Comfort Models} \tabularnewline
 \toprule 
-Mathematical variable & Description & Units ~ & Range & FORTRAN variable \tabularnewline
+Mathematical variable & Description & Units ~ & Range & Source Code Variable \tabularnewline
```
:+1:

Myoldmopar commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -50,21 +52,21 @@ \subsection{Background on Thermal Comfort Models}\label{background-on-thermal-co
 \caption{General Nomenclature list for Thermal Comfort Models \label{table:general-nomenclature-list-for-thermal-comfort}} \tabularnewline
 \toprule 
-Mathematical variable & Description & Units ~ & Range & FORTRAN variable \tabularnewline
+Mathematical variable & Description & Units ~ & Range & Source Code Variable \tabularnewline
```
:+1:

Myoldmopar commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -196,63 +196,91 @@ \subsubsection{Fanger Model Nomenclature List}\label{fanger-model-nomenclature-l
-\(L = {Q_{res}} + {Q_{dry}} + {E_{sk}} + W\) ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ W/m\(^{2}\)
+\begin{equation}
+M = L
+\end{equation}
```
Is it worth putting these units in this section somewhere?

Myoldmopar commented on doc/engineering-reference/src/simulation-models-encyclopedic-reference-005/occupant-thermal-comfort.tex:
```diff
@@ -336,163 +364,224 @@ \subsubsection{Pierce Two-Node Model Nomenclature List}\label{pierce-two-node-mo
+CloSurfTemp = (CloCond*SkinTemp + CloBodyRat*(Hc*AirTemp+Hr*RadTemp))
 
-Total evaporative heat loss from the skin, E\(_{sk}\), includes evaporation of water produced by regulatory sweating, E\(_{rsw}\), and evaporation of water vapor that diffuses through the skin surface, E\(_{diff}\).
+~~~~~~~~~~~~~~~~~~~~~~~/(CloCond + CloBodyRat*(Hc+Hr))
```
Is this supposed to be attached to the equation two lines up?  If so, can you clean up the connection?  Or just make it a real equation and not a source code representation?

RKStrand replied:
@Myoldmopar Well, here's what I can say about this.  When I delete the blank line between these two lines, it merges them back together like there is no paragraph marker between them and it messes up the spacing upon building the Eng Ref on my machine.  Same thing with PMVET, PMVSET, and DISC equations elsewhere in this section.  So, I left them...or is there another way to do this?

Myoldmopar replied:
It's fine.  I mean, really, the right thing to do, is to put them in actual equation format, instead of just pasted in code equations.  Then LaTeX can handle this formatting stuff and we don't have to keep things lined up just right.  But I get the original intent of this section and other parts of the code were to show the actual equations from the source, so just leave them.

RKStrand replied:
@Myoldmopar Yah, I think for consistency and differentiation I was using equation format for actual mathematical equations and then leaving code as text.  At least that was my intent.  Trying to get the issue that @michael-okeefe noted sorted out.  Some of these codes are a bit temperamental...but it is pretty cool that we can get all of the documentation in version control.

Myoldmopar reviewed (commented):
This is a whole bunch of cleanup, thanks for that.

Only a couple real comments:
 - There are several places in your changes where there had previously been units labeled on variables, and I think those just went away.  It's probably worth either putting the units back in the text around the variable or putting a little table/list showing the units for different terms.  Or just let me know why you think they aren't needed and we can debate.
 - I started marking the equations that were put in the docs "from the source code", because they look real funny in the docs here.  I thought there would only be one or two.  I realized after seeing more that it may be a bigger ask to clean them all up.  Can you just comment back on whether it is valuable to leave them in there like they currently are?  

I'll mark my review as a "comment only" for now, not approve or reject, so we can just chat about it before deciding whether to merge or not.

RKStrand commented:
@Myoldmopar Thanks for the thorough review!  We can definitely discuss things further though I have limited time right now.  Basically, I took all the units out of those equations because the units are contained in the tables that have all of the variable names in and because my sense of getting everything to look the same was violated by all of the floating units.  I converted all of those equations to actual equations and the units and equation numbers also looked awkward.  So, my solution was to get rid of the units since they were already in the tables, but we can discuss that further if you don't agree with that.
As for the source code equations, I'm not sure they are super useful.  It would be relatively easy to just eliminate them at this point.  Having them in there helps someone trying to follow the code, but in all honesty, those long tables are there to help with that as well.  I don't find the source equations all that useful because it detracts from the general flow of the discussion and it gets tedious.  I left them in only because they were there and I didn't feel like on my own I could just get rid of them.  However, if you also think they don't provide anything really useful, then I will just take them out.  What do you think?

RKStrand committed:
Fix of bad characters

As per review comments, fixed some bad characters in the thermal
comfort file.  Should be good to go now.

Myoldmopar commented:
Just happened to notice Linux was nicely failing to build the docs because of a math delimiter problem.  I pulled it locally, and it was actually succeeding to build the pdf, but the math was probably wonky because there's 84 LaTeX Error: Bad math environment delimiter instances!  I'm going to fix them, hijacking this branch along the way.  Once I get them cleaned I'll merge this branch in @RKStrand.  Thanks for your changes.

Myoldmopar committed:
Fix bad math delimiters

Myoldmopar commented:
OK, so it actually was the changes in this branch causing the bad math delimiters.  Just a quick heads up here.  The `\begin{equation}` opens up a math mode environment.  The `\(` command also opens up math mode.  LaTeX isn't happy when you try to open up math mode within math mode, so it throws the error.  Apparently the PDF will still build OK, but the CI dashboard catches the error and shows it as red.  

To fix the errors, I assumed that when I found `\(` inside of a `\begin{equation}`, it was intended to be an actual parenthesis, not re-opening math mode.  So I changed it to explicitly create left and right parentheses as [here](https://github.com/NREL/EnergyPlus/pull/6317/commits/318a6146dd148947b7870c8f02973181f9c9e014#diff-a581ba233e4e964a6008aabf8353bc8cR242).

Once I cleaned those up, there were no more bad math environment delimiter errors, so I'm calling this done and merging it.

RKStrand commented:
@Myoldmopar Ah...I didn't realize that both of those things were opening the math environment.  I guess I understand why it would have an issue.  Thanks for fixing that!  I'll try to remember that for any future documentation work.
